### PR TITLE
SFx exporter: add cpu delta translations

### PR DESCRIPTION
**Pull request by pmcollins**

### CODE REVIEW

File: exporter/signalfxexporter/translation/constants.go [hunk 1]
```diff
@@ -138,6 +138,37 @@ translation_rules:
...
+  mapping:
+    system.cpu.time: system.cpu.delta
+- action: copy_metrics
+  mapping:
+    system.cpu.delta: system.cpu.usage
+  dimension_key: state
+  dimension_values:
+    user: true
+    system: true
+    interrupt: true
```
Comment (dmitryax) [inline exporter/signalfxexporter/translation/constants.go:152]: there must be more states to be aggregated except for idle:
interrupt
nice
softirq
steal
system
user
wait

File: exporter/signalfxexporter/translation/constants.go [hunk 2]
```diff
@@ -138,6 +138,37 @@ translation_rules:
     receive: pod_network_receive_errors_total
     transmit: pod_network_transmit_errors_total
 
+# cpu deltas
```
Comment (dmitryax) [inline exporter/signalfxexporter/translation/constants.go:141]: let's say `compute cpu.utilization`

File: exporter/signalfxexporter/translation/constants.go [hunk 3]
```diff
@@ -138,6 +138,37 @@ translation_rules:
...
+  dimension_key: state
+  dimension_values:
+    user: true
+    system: true
+    interrupt: true
+- action: aggregate_metric
+  metric_name: system.cpu.usage
+  aggregation_method: sum
+  without_dimensions:
+  - state
```
Comment (dmitryax) [inline exporter/signalfxexporter/translation/constants.go:157]: need to add `cpu` dimension here as well

File: exporter/signalfxexporter/translation/constants.go [hunk 4]
```diff
@@ -138,6 +138,37 @@ translation_rules:
...
+- action: copy_metrics
+  mapping:
+    system.cpu.delta: system.cpu.total
+- action: aggregate_metric
+  metric_name: system.cpu.total
+  aggregation_method: sum
+  without_dimensions:
+  - state
```
Comment (dmitryax) [inline exporter/signalfxexporter/translation/constants.go:165]: Same. + `cpu`

File: exporter/signalfxexporter/translation/constants.go [hunk 5]
```diff
@@ -138,6 +138,37 @@ translation_rules:
...
+- action: calculate_new_metric
+  metric_name: cpu.utilization
+  operand1_metric: system.cpu.usage
+  operand2_metric: system.cpu.total
+  operator: /
```
Comment (dmitryax) [inline exporter/signalfxexporter/translation/constants.go:170]: let's remove `system.cpu.usage` and `system.cpu.total` after that

Comment (pmcollins) [reply exporter/signalfxexporter/translation/constants.go:176]: Done. Removed `system.cpu.delta` as well.

File: exporter/signalfxexporter/translation/constants.go (hunk 2 above)
Comment (pmcollins) [reply exporter/signalfxexporter/translation/constants.go]: Done.

File: exporter/signalfxexporter/translation/constants.go (hunk 1 above)
Comment (pmcollins) [reply exporter/signalfxexporter/translation/constants.go:150]: Done.

File: exporter/signalfxexporter/translation/constants.go (hunk 3 above)
Comment (pmcollins) [reply exporter/signalfxexporter/translation/constants.go:161]: Done.

File: exporter/signalfxexporter/translation/constants.go (hunk 4 above)
Comment (pmcollins) [reply exporter/signalfxexporter/translation/constants.go:170]: Done.

**Review by dmitryax — COMMENTED** [review]
please remove `k8s.node.cpu.utilization: cpu.utilization` mapping from translations

**Comment by pmcollins** [comment]
> please remove `k8s.node.cpu.utilization: cpu.utilization` mapping from translations

Done.

File: exporter/signalfxexporter/translation/constants.go [hunk 6]
```diff
@@ -138,6 +137,43 @@ translation_rules:
...
+  mapping:
+    system.cpu.delta: system.cpu.total
+- action: aggregate_metric
+  metric_name: system.cpu.total
+  aggregation_method: sum
+  without_dimensions:
+  - state
+  - cpu
+- action: calculate_new_metric
+  metric_name: cpu.utilization
```
Comment (dmitryax) [inline exporter/signalfxexporter/translation/constants.go:172]: Please scale the result to percents after this

**Review by dmitryax — APPROVED** [review]

**Pull request merged**